webpack/assets/javascripts/react_app/components/PF4/TableIndexPage/Table/TableHooks.js

```diff
@@ -0,0 +1,302 @@
+import { useState, useRef, useEffect, useCallback, useMemo } from 'react';
+import { isEmpty } from 'lodash';
+import { useLocation } from 'react-router-dom';
+
+class ReactConnectedSet extends Set {
+  constructor(initialValue, forceRender) {
+    super();
+    this.forceRender = forceRender;
+    // The constructor would normally call add() with the initial value, but since we
+    // must call super() at the top, this.forceRender() isn't defined yet.
+    // So, we call super() above with no argument, then call add() manually below
+    // after forceRender is defined
+    if (initialValue) {
+      if (initialValue.constructor.name === 'Array') {
+        initialValue.forEach(id => super.add(id));
+      } else {
+        super.add(initialValue);
+      }
+    }
+  }
+
+  add(value) {
+    const result = super.add(value); // ensuring these methods have the same API as the superclass
+    this.forceRender();
+    return result;
+  }
+
+  clear() {
+    const result = super.clear();
+    this.forceRender();
+    return result;
+  }
+
+  delete(value) {
+    const result = super.delete(value);
+    this.forceRender();
+    return result;
+  }
+
+  onToggle(isOpen, id) {
+    if (isOpen) {
+      this.add(id);
+    } else {
+      this.delete(id);
+    }
+  }
+
+  addAll(ids) {
+    ids.forEach(id => super.add(id));
+    this.forceRender();
+  }
+}
+
+export const useSet = initialArry => {
+  const [, setToggle] = useState(Date.now());
+  // needed because mutating a Ref won't cause React to rerender
+  const forceRender = () => setToggle(Date.now());
```

Comment:
Saw this on the internet somewhere. [Symbols](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol) are always unique values, so this may be a bit more elegant :)

```suggestion
  const forceRender = () => setToggle(Symbol());
```